mediorum/server/serve_upload.go

```diff
@@ -335,102 +339,102 @@ func (ss *MediorumServer) getBlobByJobIDAndVariant(c echo.Context) error {
 		c.Response().Header().Set("Content-Disposition", fmt.Sprintf(`attachment; filename="%s"`, contentDisposition))
 	}
 
-	jobID := c.Param("jobID")
-	variant := c.Param("variant")
-	if cidutil.IsLegacyCID(jobID) {
-		// check if file is migrated first - it would have a single key and no upload object in the db
-		key := jobID + "/" + variant
-		startMigratedExists := time.Now()
-		exists, err := ss.bucket.Exists(c.Request().Context(), key)
-		c.Response().Header().Set("x-migrated-exists", fmt.Sprintf("%.2fs", time.Since(startMigratedExists).Seconds()))
+	// 1. resolve ulid to upload cid
+	if _, err := ulid.Parse(jobID); err == nil {
+		var upload Upload
+		err := ss.crud.DB.First(&upload, "id = ?", jobID).Error
+		if err == nil {
+			// todo: cache this similar to redirectCache
+			jobID = upload.OrigFileCID
+		}
+	}
+
+	// 2. if is ba ... serve variant
+	if strings.HasPrefix(jobID, "ba") {
+		baCid := jobID
+		variantStoragePath := cidutil.ImageVariantPath(baCid, variant)
+
+		// we already have this version
+		if blob, err := ss.bucket.NewReader(ctx, variantStoragePath, nil); err == nil {
+			http.ServeContent(c.Response(), c.Request(), variantStoragePath, blob.ModTime(), blob)
+			return blob.Close()
+		}
+
+		w, h, err := parseVariantSize(variant)
```

Comment:
not that the client is requesting original.jpg anyway, but wouldn't it be good to not error here if the variant is original.jpg? then other nodes could still request it via this route for resizing or debugging